--- orig/.gitmodules	2015-10-17 18:00:33.673741323 +0200
+++ patched/.gitmodules	2015-10-17 18:01:01.130257324 +0200
@@ -1,7 +1,7 @@
 [submodule "src/llvm"]
 	path = src/llvm
-	url = https://github.com/rust-lang/llvm.git
-	branch = master
+	url = https://github.com/zazdxscf/llvm.git
+	branch = 62ad301_and_patch_for_rust_issue_28947
 [submodule "src/compiler-rt"]
 	path = src/compiler-rt
 	url = https://github.com/rust-lang/compiler-rt.git
