diff --git a/configure b/configure
index 60d3661..7cf274e 100755
--- a/configure
+++ b/configure
@@ -1324,7 +1358,8 @@ do
     make_dir $t/rt/compiler-rt
     for i in                                          \
       isaac sync test \
-      arch/i386 arch/x86_64 arch/arm arch/aarch64 arch/mips arch/powerpc
+      arch/x86_64
+      #arch/i386 arch/x86_64 arch/arm arch/aarch64 arch/mips arch/powerpc
     do
       make_dir $t/rt/stage$s/$i
     done
@@ -1551,7 +1586,8 @@ do
 
         msg "configuring LLVM for $gnu_t"
 
-        LLVM_TARGETS="--enable-targets=x86,x86_64,arm,aarch64,mips,powerpc"
+        #LLVM_TARGETS="--enable-targets=x86,x86_64,arm,aarch64,mips,powerpc"
+        LLVM_TARGETS="--enable-targets=x86_64"
         LLVM_BUILD="--build=$gnu_t"
         LLVM_HOST="--host=$gnu_t"
         LLVM_TARGET="--target=$gnu_t"
diff --git a/mk/main.mk b/mk/main.mk
index 3e3b821..ca819e5 100644
--- a/mk/main.mk
+++ b/mk/main.mk
@@ -276,8 +276,8 @@ endif
 # LLVM macros
 ######################################################################
 
-LLVM_COMPONENTS=x86 arm aarch64 mips powerpc ipo bitreader bitwriter linker asmparser mcjit \
-                interpreter instrumentation
+#LLVM_COMPONENTS=x86 arm aarch64 mips powerpc ipo bitreader bitwriter linker asmparser mcjit interpreter instrumentation
+LLVM_COMPONENTS=x86 ipo bitreader bitwriter linker asmparser mcjit interpreter instrumentation
 
 # Only build these LLVM tools
 LLVM_TOOLS=bugpoint llc llvm-ar llvm-as llvm-dis llvm-mc opt llvm-extract
diff --git a/mk/llvm.mk b/mk/llvm.mk
index 1cbf4a9..f8d5387 100644
--- a/mk/llvm.mk
+++ b/mk/llvm.mk
@@ -46,11 +46,11 @@ else
 
 $$(LLVM_CONFIG_$(1)): $$(LLVM_DEPS) $$(LLVM_STAMP_$(1))
 	@$$(call E, make: llvm)
-	$$(Q)$$(MAKE) -C $$(CFG_LLVM_BUILD_DIR_$(1)) $$(CFG_LLVM_BUILD_ENV_$(1)) ONLY_TOOLS="$$(LLVM_TOOLS)"
+	$$(Q)$$(MAKE) -j8 -C $$(CFG_LLVM_BUILD_DIR_$(1)) $$(CFG_LLVM_BUILD_ENV_$(1)) ONLY_TOOLS="$$(LLVM_TOOLS)"
 	$$(Q)touch $$(LLVM_CONFIG_$(1))
 
 clean-llvm$(1):
-	$$(Q)$$(MAKE) -C $$(CFG_LLVM_BUILD_DIR_$(1)) clean
+	$$(Q)$$(MAKE) -j8 -C $$(CFG_LLVM_BUILD_DIR_$(1)) clean
 
 endif
 
diff --git a/src/librustc_llvm/lib.rs b/src/librustc_llvm/lib.rs
index fe84cff..851b349 100644
--- a/src/librustc_llvm/lib.rs
+++ b/src/librustc_llvm/lib.rs
@@ -2014,7 +2014,7 @@ extern {
     pub fn LLVMInitializeX86TargetMC();
     pub fn LLVMInitializeX86AsmPrinter();
     pub fn LLVMInitializeX86AsmParser();
-    pub fn LLVMInitializeARMTargetInfo();
+/*    pub fn LLVMInitializeARMTargetInfo();
     pub fn LLVMInitializeARMTarget();
     pub fn LLVMInitializeARMTargetMC();
     pub fn LLVMInitializeARMAsmPrinter();
@@ -2034,7 +2034,7 @@ extern {
     pub fn LLVMInitializePowerPCTargetMC();
     pub fn LLVMInitializePowerPCAsmPrinter();
     pub fn LLVMInitializePowerPCAsmParser();
-
+*/
     pub fn LLVMRustAddPass(PM: PassManagerRef, Pass: *const c_char) -> bool;
     pub fn LLVMRustCreateTargetMachine(Triple: *const c_char,
                                        CPU: *const c_char,
diff --git a/src/librustc_trans/back/write.rs b/src/librustc_trans/back/write.rs
index 1fbbf82..933a262 100644
--- a/src/librustc_trans/back/write.rs
+++ b/src/librustc_trans/back/write.rs
@@ -967,7 +967,7 @@ pub unsafe fn configure_llvm(sess: &Session) {
     llvm::LLVMInitializeX86AsmPrinter();
     llvm::LLVMInitializeX86AsmParser();
 
-    llvm::LLVMInitializeARMTargetInfo();
+/*    llvm::LLVMInitializeARMTargetInfo();
     llvm::LLVMInitializeARMTarget();
     llvm::LLVMInitializeARMTargetMC();
     llvm::LLVMInitializeARMAsmPrinter();
@@ -990,7 +990,7 @@ pub unsafe fn configure_llvm(sess: &Session) {
     llvm::LLVMInitializePowerPCTargetMC();
     llvm::LLVMInitializePowerPCAsmPrinter();
     llvm::LLVMInitializePowerPCAsmParser();
-
+*/
     llvm::LLVMRustSetLLVMOptions(llvm_args.len() as c_int,
                                  llvm_args.as_ptr());
 }
