diff -pru5 -X /home/emacs/build/kernel/linuxgit/makepkg/linux-git/src/linux-git/Documentation/dontdiff -- xfce4-weather-plugin.orig/configure.ac.in xfce4-weather-plugin/configure.ac.in
--- xfce4-weather-plugin.orig/configure.ac.in	2014-11-26 01:39:12.196996518 +0100
+++ xfce4-weather-plugin/configure.ac.in	2014-11-26 01:53:03.215769680 +0100
@@ -18,11 +18,11 @@ m4_define([weather_version], [weather_ve
 dnl ***************************
 dnl *** Initialize autoconf ***
 dnl ***************************
 AC_COPYRIGHT([Copyright (c) 2006-2014
         The Xfce development team. All rights reserved.])
-AC_INIT([xfce4-weather-plugin], [weather_version], [http://bugzilla.xfce.org/], [xfce4-weather-plugin])
+AC_INIT([xfce4-weather-plugin], [weather_version], [https://bugzilla.xfce.org/], [xfce4-weather-plugin])
 AC_PREREQ([2.60])
 AC_CONFIG_MACRO_DIR([m4])
 
 dnl ***************************
 dnl *** Initialize automake ***
diff -pru5 -X /home/emacs/build/kernel/linuxgit/makepkg/linux-git/src/linux-git/Documentation/dontdiff -- xfce4-weather-plugin.orig/panel-plugin/weather.c xfce4-weather-plugin/panel-plugin/weather.c
--- xfce4-weather-plugin.orig/panel-plugin/weather.c	2014-11-26 01:39:12.233662983 +0100
+++ xfce4-weather-plugin/panel-plugin/weather.c	2014-11-26 01:49:58.063452360 +0100
@@ -617,11 +617,11 @@ update_handler(plugin_data *data)
         /* calculate date range for request */
         end_t = time_calc_day(now_tm, ASTRODATA_MAX_DAYS);
         end_tm = *localtime(&end_t);
 
         /* build url */
-        url = g_strdup_printf("http://api.yr.no/weatherapi/sunrise/1.0/?"
+        url = g_strdup_printf("http://api.yr.no/weatherapi/sunrise/1.0/?" //https ERR_CONNECTION_REFUSED
                               "lat=%s;lon=%s;"
                               "from=%04d-%02d-%02d;"
                               "to=%04d-%02d-%02d",
                               data->lat, data->lon,
                               now_tm.tm_year + 1900,
@@ -644,11 +644,11 @@ update_handler(plugin_data *data)
         data->weather_update->next = time_calc_hour(now_tm, 1);
         data->weather_update->started = TRUE;
 
         /* build url */
         url =
-            g_strdup_printf("http://api.yr.no/weatherapi"
+            g_strdup_printf("http://api.yr.no/weatherapi" //no https port open
                             "/locationforecastlts/1.2/?lat=%s;lon=%s;msl=%d",
                             data->lat, data->lon, data->msl);
 
         /* start receive thread */
         g_message(_("getting %s"), url);
diff -pru5 -X /home/emacs/build/kernel/linuxgit/makepkg/linux-git/src/linux-git/Documentation/dontdiff -- xfce4-weather-plugin.orig/panel-plugin/weather-config.c xfce4-weather-plugin/panel-plugin/weather-config.c
--- xfce4-weather-plugin.orig/panel-plugin/weather-config.c	2014-11-26 01:39:12.230329668 +0100
+++ xfce4-weather-plugin/panel-plugin/weather-config.c	2014-11-26 01:47:45.780844920 +0100
@@ -292,21 +292,21 @@ lookup_altitude_timezone(const gpointer
 
     latstr = double_to_string(lat, "%.6f");
     lonstr = double_to_string(lon, "%.6f");
 
     /* lookup altitude */
-    url = g_strdup_printf("http://api.geonames.org"
+    url = g_strdup_printf("http://api.geonames.org" //ERR_SSL_PROTOCOL_ERROR
                           "/srtm3XML?lat=%s&lng=%s&username=%s",
                           latstr, lonstr,
                           dialog->pd->geonames_username
                           ? dialog->pd->geonames_username : GEONAMES_USERNAME);
     weather_http_queue_request(dialog->pd->session, url,
                                cb_lookup_altitude, user_data);
     g_free(url);
 
     /* lookup timezone */
-    url = g_strdup_printf("http://api.geonames.org"
+    url = g_strdup_printf("http://api.geonames.org" //SSL connection error
                           "/timezone?lat=%s&lng=%s&username=%s",
                           latstr, lonstr,
                           dialog->pd->geonames_username
                           ? dialog->pd->geonames_username : GEONAMES_USERNAME);
     weather_http_queue_request(dialog->pd->session, url,
diff -pru5 -X /home/emacs/build/kernel/linuxgit/makepkg/linux-git/src/linux-git/Documentation/dontdiff -- xfce4-weather-plugin.orig/panel-plugin/weather.h xfce4-weather-plugin/panel-plugin/weather.h
--- xfce4-weather-plugin.orig/panel-plugin/weather.h	2014-11-26 01:39:12.233662983 +0100
+++ xfce4-weather-plugin/panel-plugin/weather.h	2014-11-26 01:51:37.492906746 +0100
@@ -25,11 +25,11 @@
 #ifdef HAVE_UPOWER_GLIB
 #include <upower.h>
 #endif
 #include "weather-icon.h"
 
-#define PLUGIN_WEBSITE "http://goodies.xfce.org/projects/panel-plugins/xfce4-weather-plugin"
+#define PLUGIN_WEBSITE "http://goodies.xfce.org/projects/panel-plugins/xfce4-weather-plugin" //https cert is for different url
 #define MAX_FORECAST_DAYS 10
 #define DEFAULT_FORECAST_DAYS 5
 #define MAX_SCROLLBOX_LINES 10
 
 G_BEGIN_DECLS
diff -pru5 -X /home/emacs/build/kernel/linuxgit/makepkg/linux-git/src/linux-git/Documentation/dontdiff -- xfce4-weather-plugin.orig/panel-plugin/weather-search.c xfce4-weather-plugin/panel-plugin/weather-search.c
--- xfce4-weather-plugin.orig/panel-plugin/weather-search.c	2014-11-26 01:39:12.230329668 +0100
+++ xfce4-weather-plugin/panel-plugin/weather-search.c	2014-11-26 01:41:53.992775339 +0100
@@ -159,11 +159,11 @@ search_cb(GtkWidget *widget,
 
     gtk_widget_set_sensitive(dialog->find_button, FALSE);
     gtk_dialog_set_response_sensitive(GTK_DIALOG(dialog->dialog),
                                       GTK_RESPONSE_ACCEPT, FALSE);
 
-    url = g_strdup_printf("http://nominatim.openstreetmap.org/"
+    url = g_strdup_printf("https://nominatim.openstreetmap.org/"
                           "search?q=%s&format=xml", sane_str);
     g_free(sane_str);
 
     gtk_tree_view_column_set_title(dialog->column, _("Searching..."));
     g_message(_("getting %s"), url);
@@ -421,11 +421,11 @@ void weather_search_by_ip(SoupSession *s
                                           const units_config *units,
                                           gpointer user_data),
                           gpointer user_data)
 {
     geolocation_data *data;
-    const gchar *url = "http://geoip.xfce.org/";
+    const gchar *url = "http://geoip.xfce.org/"; //https is self issue cert for diff url
 
     if (!gui_cb)
         return;
 
     data = g_new0(geolocation_data, 1);
diff -pru5 -X /home/emacs/build/kernel/linuxgit/makepkg/linux-git/src/linux-git/Documentation/dontdiff -- xfce4-weather-plugin.orig/panel-plugin/weather-summary.c xfce4-weather-plugin/panel-plugin/weather-summary.c
--- xfce4-weather-plugin.orig/panel-plugin/weather-summary.c	2014-11-26 01:39:12.230329668 +0100
+++ xfce4-weather-plugin/panel-plugin/weather-summary.c	2014-11-26 01:45:34.314899666 +0100
@@ -302,11 +302,11 @@ weather_summary_get_logo(plugin_data *da
 
     pixbuf = gdk_pixbuf_new_from_file(path, NULL);
     g_free(path);
     if (pixbuf == NULL)
         weather_http_queue_request(data->session,
-                                   "http://met.no/filestore/met.no-logo.gif",
+                                   "http://met.no/filestore/met.no-logo.gif", //diff url cert for https
                                    logo_fetched, image);
     else {
         gtk_image_set_from_pixbuf(GTK_IMAGE(image), pixbuf);
         g_object_unref(pixbuf);
     }
@@ -546,17 +546,17 @@ create_summary_tab(plugin_data *data)
     ltag_geonames = gtk_text_buffer_create_tag(buffer, "lnk3",
                                                "foreground-gdk",
                                                &lnk_color, NULL);
     APPEND_BTEXT(_("\nCredits\n"));
     APPEND_LINK_ITEM(_("\tEncyclopedic information partly taken from\n\t\t"),
-                     _("Wikipedia"), "http://wikipedia.org", ltag_wiki);
+                     _("Wikipedia"), "https://wikipedia.org", ltag_wiki);
     APPEND_LINK_ITEM(_("\n\tElevation and timezone data provided by\n\t\t"),
                      _("GeoNames"),
-                     "http://geonames.org/", ltag_geonames);
+                     "http://geonames.org/", ltag_geonames); //no https
     APPEND_LINK_ITEM(_("\n\tWeather and astronomical data from\n\t\t"),
                      _("The Norwegian Meteorological Institute"),
-                     "http://met.no/", ltag_metno);
+                     "http://met.no/", ltag_metno); //https cert is for diff url
     g_object_set_data_full(G_OBJECT(ltag_img), "url",   /* url for image */
                            g_strdup("http://met.no"), g_free);
 
     g_signal_connect(G_OBJECT(view), "motion-notify-event",
                      G_CALLBACK(view_motion_notify), sum);
