diff --git a/src/librustdoc/html/render.rs b/src/librustdoc/html/render.rs
index 0f436ef..c342c97 100644
--- a/src/librustdoc/html/render.rs
+++ b/src/librustdoc/html/render.rs
@@ -449,7 +449,9 @@ pub fn run(mut krate: clean::Crate,
                 }
                 clean::NameValue(ref x, ref s)
                         if "html_playground_url" == *x => {
+                            println!("html_playground_url(pre )={}", &cx.layout.playground_url);
                     cx.layout.playground_url = s.to_string();
+                            println!("html_playground_url(post)={}", &cx.layout.playground_url);
                     markdown::PLAYGROUND_KRATE.with(|slot| {
                         if slot.borrow().is_none() {
                             let name = krate.name.clone();
