--- asorted	2014-08-09 20:46:57.157408285 +0200
+++ bsorted	2014-08-09 20:47:01.424050060 +0200
@@ -30,12 +30,10 @@ URxvt*troughColor:	[90]#FF0000
 URxvt.url-select.launcher:	chromium-dev
 URxvt.url-select.underline:	true
 URxvt*visualBell:	true
-Xcursor.size:	0
-Xcursor.theme:	Chameleon-White-Small
-Xcursor.theme_core:	1
-Xft.antialias:	1
+Xft.antialias:	true
 Xft.autohint:	false
 Xft.dpi:	96
-Xft.hintstyle:	hintfull
+Xft.hinting:	true
+Xft.hintstyle:	hintslight
 Xft.lcdfilter:	lcddefault
 Xft.rgba:	rgb
