--- "archlinux version!_dkms-2.2.0.3-15-any/dkms"	2014-11-05 11:41:37.000000000 +0100
+++ "manjaro(archlinux based) version patched_dkms-2.2.0.3-15-any/dkms"	2014-11-05 11:47:08.029088432 +0100
@@ -457,11 +457,11 @@ read_conf()
     # Find which conf file to check
     [[ $conf ]] && read_conf_file="$conf"
     [[ $3 ]] && read_conf_file="$3"
 
     [[ -r $read_conf_file ]] || die 4 $"Could not locate dkms.conf file." \
-	$"File: $conf does not exist."
+	$"File: $read_conf_file does not exist."
 
     [[ $last_mvka = $module/$module_version/$1/$2 && \
 	$last_mvka_conf = $(readlink -f $read_conf_file) ]] && return
 
 
