diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 78b4bad10..1063540 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -2338,8 +2338,8 @@ EXPORT_SYMBOL_GPL(preempt_notifier_dec);
  */
 void preempt_notifier_register(struct preempt_notifier *notifier)
 {
-	if (!static_key_false(&preempt_notifier_key))
-		WARN(1, "registering preempt_notifier while notifiers disabled\n");
+	if (!static_key_false(&preempt_notifier_key));;;
+//		WARN(1, "registering preempt_notifier while notifiers disabled\n");
 
 	hlist_add_head(&notifier->link, &current->preempt_notifiers);
 }
