--- /dev/fd/63	2015-01-15 23:33:17.236038870 +0100
+++ /dev/fd/62	2015-01-15 23:33:17.236038870 +0100
@@ -16 +16 @@ Release Date: 2014-12-20
- (==) Log file: "/home/emacs/.local/share/xorg/Xorg.0.log", Time: Thu Jan 15 23:21:36 2015
+ (==) Log file: "/home/emacs/.local/share/xorg/Xorg.0.log", Time: Thu Jan 15 23:21:51 2015
@@ -661,24 +660,0 @@ Release Date: 2014-12-20
- (II) UnloadModule: "synaptics"
- (II) systemd-logind: releasing fd for 13:70
- (II) evdev: AT Translated Set 2 keyboard: Close
- (II) UnloadModule: "evdev"
- (II) systemd-logind: releasing fd for 13:69
- (II) evdev: Ideapad extra buttons: Close
- (II) UnloadModule: "evdev"
- (II) systemd-logind: releasing fd for 13:72
- (II) evdev: Genius 2.4G Wireless BlueEye Mouse: Close
- (II) UnloadModule: "evdev"
- (II) systemd-logind: releasing fd for 13:71
- (II) evdev: Sleep Button: Close
- (II) UnloadModule: "evdev"
- (II) systemd-logind: releasing fd for 13:66
- (II) evdev: Power Button: Close
- (II) UnloadModule: "evdev"
- (II) systemd-logind: releasing fd for 13:64
- (II) evdev: Video Bus: Close
- (II) UnloadModule: "evdev"
- (II) systemd-logind: releasing fd for 13:68
- (II) evdev: Power Button: Close
- (II) UnloadModule: "evdev"
- (II) systemd-logind: releasing fd for 13:67
- (EE) Server terminated successfully (0). Closing log file.
