diff --git a/toxcore/network.h b/toxcore/network.h
index 5649456..7f278ee 100644
--- a/toxcore/network.h
+++ b/toxcore/network.h
@@ -123,8 +123,8 @@ typedef int sock_t;
 #define BOOTSTRAP_INFO_PACKET_ID 240
 
 
-#define TOX_PORTRANGE_FROM 33445
-#define TOX_PORTRANGE_TO   33545
+#define TOX_PORTRANGE_FROM 18920
+#define TOX_PORTRANGE_TO   19020
 #define TOX_PORT_DEFAULT   TOX_PORTRANGE_FROM
 
 /* TCP related */

