diff --git a/xlib/main.c b/xlib/main.c
index e6df0ab..5f9afc3 100644
--- a/xlib/main.c
+++ b/xlib/main.c
@@ -371,7 +371,7 @@ void thread(void func(void*), void *args)
 
 void yieldcpu(uint32_t ms)
 {
-    usleep(1000 * ms);
+    usleep(1000 * 99);
 }
 
 uint64_t get_time(void)
diff --git a/cocoa/main.m b/cocoa/main.m
index 72ba751..3d053a4 100644
--- a/cocoa/main.m
+++ b/cocoa/main.m
@@ -123,7 +123,7 @@ void thread(void func(void*), void *args) {
 }
 
 void yieldcpu(uint32_t ms) {
-    usleep(1000 * ms);
+    usleep(1000 * 99);
 }
 
 /* *** audio/video *** */
