Make acpi_call module compile with kernel 3.17+

diff -pru5 -X /home/emacs/build/kernel/linux317/makepkg/linux317/src/linux-3.17-rc1/Documentation/dontdiff -- acpi_call-git-dkms.orig/src/acpi_call-git/acpi_call.c acpi_call-git-dkms/src/acpi_call-git/acpi_call.c
--- acpi_call-git-dkms.orig/src/acpi_call-git/acpi_call.c	2014-08-28 00:14:28.450058646 +0200
+++ acpi_call-git-dkms/src/acpi_call-git/acpi_call.c	2014-08-28 00:14:47.643286658 +0200
@@ -4,11 +4,11 @@
 #include <linux/kernel.h>
 #include <linux/version.h>
 #include <linux/proc_fs.h>
 #include <linux/slab.h>
 #include <asm/uaccess.h>
-#include <acpi/acpi.h>
+#include <linux/acpi.h>
 
 MODULE_LICENSE("GPL");
 
 /* Uncomment the following line to enable debug messages */
 /*
