Index: grub-9999/configure.ac
===================================================================
--- grub-9999.orig/configure.ac
+++ grub-9999/configure.ac
@@ -267,7 +267,7 @@ else
      netbsd* | openbsd*)
         # Because /boot is used for the boot block in NetBSD and OpenBSD,
            bootdirname=''      ;;
-     *)    bootdirname='boot'  ;;
+     *)    bootdirname='bewt'  ;;
    esac
 fi
 
