diff --git a/third_party/WebKit/Source/bindings/scripts/BUILD.gn b/third_party/WebKit/Source/bindings/scripts/BUILD.gn
index 36b5cf1..6221820 100644
--- a/third_party/WebKit/Source/bindings/scripts/BUILD.gn
+++ b/third_party/WebKit/Source/bindings/scripts/BUILD.gn
@@ -32,7 +32,7 @@ action("cached_lex_yacc_tables") {
 action("cached_jinja_templates") {
   script = "code_generator.py"
 
-  inputs = jinja_module_files + [ "code_generator.py" ] +
+  inputs = [ "code_generator.py" ] +
            code_generator_template_files
 
   # Dummy file to track dependency.
