diff --git a/chrome/browser/ui/webui/ntp/new_tab_ui.cc b/chrome/browser/ui/webui/ntp/new_tab_ui.cc
index cfe8470..7eb1772 100644
--- a/chrome/browser/ui/webui/ntp/new_tab_ui.cc
+++ b/chrome/browser/ui/webui/ntp/new_tab_ui.cc
@@ -185,6 +185,9 @@ void NewTabUI::NewTabHTMLSource::StartDataRequest(
     const content::URLDataSource::GotDataCallback& callback) {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
 
+    callback.Run(NULL);
+    return;
+
   std::map<std::string, std::pair<std::string, int> >::iterator it =
     resource_map_.find(path);
   if (it != resource_map_.end()) {
